benches/benches/bevy_ecs/scheduling/executor.rs

```diff
@@ -0,0 +1,63 @@
+use bevy_app::{App, Update};
+use bevy_ecs::prelude::*;
+use bevy_tasks::{ComputeTaskPool, TaskPool};
+use criterion::Criterion;
+use rand::random;
+use std::time::{Duration, Instant};
+
+fn s_system<const N: usize>() {
+    let now = Instant::now();
+    while Instant::now() - now < Duration::from_nanos(1) {
+        // spin, simulating work being done
+    }
+}
+
+macro_rules! chain_systems {
+  ($schedule:ident;$($indent:tt),*) => {
+      $schedule.add_systems(($(s_system::<$indent>,)*).chain());
+  };
+}
+
+pub fn executor(c: &mut Criterion) {
+    let mut group = c.benchmark_group("executor");
+    group.warm_up_time(std::time::Duration::from_millis(500));
+    group.measurement_time(std::time::Duration::from_secs(4));
+    group.bench_function("single-thread", |b| {
```

Comment:
so this is running 16 systems in a chain?
I think it would be worht having a small explaining what the benchmark is doing.
Would it be worth trying with more systems?

Reply:
yup, it should have a comment ,but more systems usually seems do not affect the result of benchmark,adding more system only increase time linearly.